### PR TITLE
create aspera sync script

### PR DESCRIPTION
**Pull request by idelcano**
### :pushpin: References
* **Issue:** Trello card: Test aspera installation to new aspera share point

### :tophat: What is the goal?

The idea is to create a script that automatically detects any change in the file share and download it. 
To decide which file or directory to download, the script compares the dates and names obtained with the ls command, against the Aspera browse table. 
If a date is before the date of the server, the file must be downloaded again.
The wrong files on the server side will be ignored.

### :memo: How is it being implemented?

I used ls to get the local info, and Aspera browser command to get the remote info.
I used the sed and awk commands to extract each file information.
 
I used the busybox date command to transform the local and server dates in seconds.

### :boom: How can it be tested?

 **Use case 1:** - Run the script in a user with Aspera properly installed.

All the files and server should be download

 **Use case 2:** - Run the script again

All it's already downloaded.

 **Use case 3:** - Add a file or create a directory using the Aspera web client, and run the script again

The modified objects will be downloaded.

### CODE REVIEW

**Comment by adrianq** [comment]
@tokland I added you as a reviewer here. I will explain tomorrow the specifics of this task (and pass you the password) so you can review it properly

File: cnb/aspera_sync.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                        localdate=`echo $localdate$localhour | cut -f1 -d"."`
+                        localfiledate=`busybox date -u -d $localdate -D %Y-%m-%d%H:%M:%S`
+                        localfiledate=`date -d "$localfiledate" +%s`
+                fi
+                filedate=`busybox date -u -d $date -D %Y-%m-%dT%H:%M:%SZ`
+                filedate=`date -d "$filedate" +%s`
+
+                #Check if the filename exist in the local filenames, and if the creation date is before than the server updated date
+                if [[ "$localfilenames" == *"$filename"* ]] && [[ $filedate -lt $localfiledate ]]; then
+                        echo "Already exist: $filename"
```
Comment (tokland) [inline cnb/aspera_sync.sh]: A question about this: I tried to download files that already existed on destination and I get: 

```
100%[======================================] 962MB / 962MB              00:00    
Files: 0 transferred, 1 skipped, 0 failed
```

So, why do we need to manually check file dates? 

Some more info: aspera seems to check the hash of the files (I changed one byte of a file and the file was synced again, as expected). I did some tests with partial files. It does not seem to do incremental sync (like rsync).

File: cnb/aspera_sync.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        lslist=`TZ=utc ls $LOCAL_DATA -altr --color=auto --time-style=full-iso`
+
+        #Get filenames column
+        localfilenames=`echo "$lslist" | awk '{print $9}'`
+
+        #Get aspera folder info and remove all less files table
+        result=`aspera shares browse -i -u $REMOTE_USER --host $REMOTE_HOST --path=$REMOTE_FOLDER  --sort mtime_a | sed -n  '/----/,/----/p' | sed '$d;1d;2d' `
+
+        echo "$result" |
+        while IFS= read -r asperarow;do
```
Comment (tokland) [inline cnb/aspera_sync.sh]: Not sure about this. We check the root files in source (aspera share) and destination (hard disk), but just the root elements (files and directories). What about nested files within directories? This is related with a note below regarding the ability to skip files of aspera.

**Comment by adrianq** [comment]
Close in favour of https://github.com/I2PC/i2pc-sys-tools/pull/2

**Pull request closed without merging**